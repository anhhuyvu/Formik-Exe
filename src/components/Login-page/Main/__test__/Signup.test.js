import { act, fireEvent } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import SignupForm from "../Signup";
import { render, unmountComponentAtNode } from "react-dom";

describe("Contact Component", () => {
  let container = null;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    document.body.appendChild(container);
  });
  afterEach(() => {
    // cleanup on exiting
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });
  it("changes value when clicked", () => {
    const onChange = jest.fn();
    act(() => {
      render(<SignupForm onChange={onChange} />, container);
    });
  });
  it("should render component correctly", () => {
    const component = renderer.create(<SignupForm></SignupForm>).toJSON();
    expect(component).toMatchSnapshot();
  });
  it("should submit correct information", async () => {
    const { getByLabelText, getByTextId } = render(<SignupForm></SignupForm>);
    const name = getByLabelText("Your Name");
    const selectValue = getByLabelText("Select");
    const message = getByLabelText("Message");
    const formContact = getByTextId("signup-form");
    const dummyInfo = {
      dummyName: "Huy Le Vu Anh",
      dummySelect: "Rider",
      dummyMessage: "This is message for testing",
    };
    act(() => {
      fireEvent.change(name, { target: { value: dummyInfo.dummyName } });
      fireEvent.change(selectValue, {
        target: { value: dummyInfo.dummySelect },
      });
      fireEvent.change(message, { target: { value: dummyInfo.dummyMessage } });
      fireEvent.submit(formContact);
    });
    await (() => {
      expect(name).toHaveTextContent(dummyInfo.fakeName);
      expect(selectValue).toHaveTextContent(dummyInfo.fakeSelect);
      expect(message).toHaveTextContent(dummyInfo.fakeMessage);
    });
  });
});
