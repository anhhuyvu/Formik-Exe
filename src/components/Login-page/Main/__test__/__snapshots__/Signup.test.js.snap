// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Component should render component correctly 1`] = `
<div
  className="content"
>
  <form
    action="#"
    id="signup-form"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="icon"
    >
      <i
        class="fa fa-envelope"
      />
      <p
        style={
          Object {
            "fontSize": "20px",
          }
        }
      >
        Contact Us
      </p>
    </div>
    <div
      className="top-content"
    >
      <div
        className="form-name-input"
      >
        <label
          htmlFor="name"
        >
          Your Name
        </label>
        <input
          className="name-input"
          data-testid="name"
          initialTouched={false}
          initialValue=""
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter the issue"
          touched={false}
          value=""
        />
      </div>
      <div
        className="form-message"
      >
        <label
          htmlFor="message"
        >
          Message
        </label>
        <textarea
          className="message"
          initialTouched={false}
          initialValue=""
          name="message"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Enter your message"
          rows={3}
          touched={false}
          value=""
        />
      </div>
    </div>
    <div
      className="bottom-content"
    >
      <div
        className="Form-Select"
      >
        <label
          htmlFor="vehicleType"
        >
          Select
        </label>
        <select
          initialTouched={false}
          initialValue="Select"
          name="vehicleType"
          onBlur={[Function]}
          onChange={[Function]}
          touched={false}
          value="Select"
        >
          <option
            value="Select"
          >
            Select
          </option>
          <option
            value="Rider"
          >
            Rider
          </option>
          <option
            value="Driver"
          >
            Driver
          </option>
        </select>
      </div>
      <button
        className="submit-button"
        disabled={true}
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
